### PR TITLE
[P4-1422] Adds ability to control creation of a court hearing in Nomis

### PR DESCRIPTION
**Pull request by willfish**
### Jira link

P4-1422

### What?

- [x] Adds query param that controls creation of court hearings in Nomis

### Why?

This is needed by the frontend to conditionally not create a court hearing in Nomis in some of their frontend flows

### Have you? (optional)

- [x] Updated API docs if necessary

### CODE REVIEW

File: spec/swagger/api/v1/court_hearings_controller_create_spec.rb [hunk 1]
```diff
@@ -23,6 +23,12 @@
           If you're testing interactively in the web UI, you can ignore this field
         DESCRIPTION
 
+      parameter name: :should_save_court_hearings_in_nomis,
+          in: :query,
+          description: 'Create the court hearing in Nomis as well as Book As A Secure Move',
```
Comment (willfish) [inline spec/swagger/api/v1/court_hearings_controller_create_spec.rb:28]:
```suggestion
          description: 'Create the court hearing in Nomis as well as Book A Secure Move',
```

File: app/controllers/api/v1/court_hearings_controller.rb [hunk 2]
```diff
@@ -31,11 +31,17 @@ def court_hearings_params
...
 
       def move
-        id = params.require(:data).dig(:relationships, :move, :data, :id)
+        @move ||= begin
+                    id = params.require(:data).dig(:relationships, :move, :data, :id)
 
-        return if id.blank?
+                    if id.present?
+                      Move.find(id)
+                    end
```
Comment (alexdesi) [inline app/controllers/api/v1/court_hearings_controller.rb:39]: is not the same as `return if id.blank?`

File: app/controllers/api/v1/court_hearings_controller.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ class CourtHearingsController < ApiController
       def create
         court_hearing = CourtHearing.create!(court_hearings_attributes)
 
-        if move
+        if should_save_court_hearings?
```
Comment (alexdesi) [inline app/controllers/api/v1/court_hearings_controller.rb:7]: `should_save_court_hearings` sounds like it should be `.save` in local,
what about
`should_save_in_nomis` or `should_be_saved_in_nomis`?

File: app/controllers/api/v1/court_hearings_controller.rb (hunk 2 above)
Comment (willfish) [reply app/controllers/api/v1/court_hearings_controller.rb]: It is, yeah :).

File: app/controllers/api/v1/court_hearings_controller.rb (hunk 3 above)
Comment (willfish) [inline app/controllers/api/v1/court_hearings_controller.rb:7]:
```suggestion
        if should_save_court_hearings_in_nomis?
```

File: app/controllers/api/v1/court_hearings_controller.rb [hunk 4]
```diff
@@ -31,11 +31,17 @@ def court_hearings_params
...
 
-        return if id.blank?
+                    return if id.blank?
 
-        Move.find(id)
+                    Move.find(id)
+                  end
+      end
+
+      def should_save_court_hearings?
```
Comment (willfish) [inline app/controllers/api/v1/court_hearings_controller.rb:43]:
```suggestion
      def should_save_court_hearings_in_nomis?
```

File: app/controllers/api/v1/court_hearings_controller.rb (hunk 3 above)
Comment (willfish) [reply app/controllers/api/v1/court_hearings_controller.rb:7]: I'll keep it consistent with the query param

File: spec/swagger/api/v1/court_hearings_controller_create_spec.rb [hunk 5]
```diff
@@ -23,6 +23,12 @@
           If you're testing interactively in the web UI, you can ignore this field
         DESCRIPTION
 
+      parameter name: :should_save_court_hearings_in_nomis,
```
Comment (willfish) [inline spec/swagger/api/v1/court_hearings_controller_create_spec.rb:26]:
```suggestion
      parameter name: : should_save_in_nomis,
```

Comment (willfish) [inline spec/swagger/api/v1/court_hearings_controller_create_spec.rb:26]:
```suggestion
      parameter name: :should_save_in_nomis,
```

File: spec/requests/api/v1/court_hearings_controller_create_spec.rb [hunk 6]
```diff
@@ -64,10 +64,34 @@
         expect(move.court_hearings).to include(court_hearing)
       end
 
-      it 'creates the court hearings in Nomis' do
-        post '/api/v1/court_hearings', params: { data: data }, headers: headers, as: :json
+      context 'when should_save_court_hearings_in_nomis param is true' do
+        let(:query_params) { '?should_save_court_hearings_in_nomis=true' }
```
Comment (willfish) [inline spec/requests/api/v1/court_hearings_controller_create_spec.rb:68]:
```suggestion
        let(:query_params) { '?should_save_in_nomis =true' }
```

File: spec/requests/api/v1/court_hearings_controller_create_spec.rb [hunk 7]
```diff
@@ -64,10 +64,34 @@
...
+
+        it 'creates the court hearings in Nomis' do
+          post "/api/v1/court_hearings#{query_params}", params: { data: data }, headers: headers, as: :json
+
+          expect(CourtHearings::CreateInNomis).to have_received(:call).with(move, move.court_hearings)
+        end
+      end
+
+      context 'when should_save_court_hearings_in_nomis param is not true' do
+        let(:query_params) { '?should_save_court_hearings_in_nomis=foo' }
```
Comment (willfish) [inline spec/requests/api/v1/court_hearings_controller_create_spec.rb:78]:
```suggestion
        let(:query_params) { '?should_save_in_nomis=foo' }
```

File: app/controllers/api/v1/court_hearings_controller.rb [hunk 8]
```diff
@@ -31,11 +31,17 @@ def court_hearings_params
...
-        return if id.blank?
+                    return if id.blank?
 
-        Move.find(id)
+                    Move.find(id)
+                  end
+      end
+
+      def should_save_court_hearings_in_nomis?
+        move && params['should_save_court_hearings_in_nomis'] == 'true'
```
Comment (willfish) [inline app/controllers/api/v1/court_hearings_controller.rb:44]:
```suggestion
        move && params['should_save_in_nomis'] == 'true'
```

File: app/controllers/api/v1/court_hearings_controller.rb [hunk 9]
```diff
@@ -4,7 +4,7 @@ class CourtHearingsController < ApiController
       def create
         court_hearing = CourtHearing.create!(court_hearings_attributes)
 
-        if move
+        if should_save_court_hearings_in_nomis?
```
Comment (willfish) [inline app/controllers/api/v1/court_hearings_controller.rb:7]:
```suggestion
        if should_save_in_nomis?
```

File: spec/requests/api/v1/court_hearings_controller_create_spec.rb [hunk 10]
```diff
@@ -64,10 +64,34 @@
...
+        let(:query_params) { '?should_save_in_nomis =true' }
+
+        it 'creates the court hearings in Nomis' do
+          post "/api/v1/court_hearings#{query_params}", params: { data: data }, headers: headers, as: :json
+
+          expect(CourtHearings::CreateInNomis).to have_received(:call).with(move, move.court_hearings)
+        end
+      end
+
+      context 'when should_save_court_hearings_in_nomis param is not true' do
```
Comment (willfish) [inline spec/requests/api/v1/court_hearings_controller_create_spec.rb:77]:
```suggestion
      context 'when should_save_in_nomis param is not true' do
```

File: spec/requests/api/v1/court_hearings_controller_create_spec.rb [hunk 11]
```diff
@@ -64,10 +64,34 @@
...
+        let(:query_params) { '?should_save_in_nomis=foo' }
+
+        it 'creates the court hearings in Nomis' do
+          post "/api/v1/court_hearings#{query_params}", params: { data: data }, headers: headers, as: :json
+
+          expect(CourtHearings::CreateInNomis).not_to have_received(:call).with(move, move.court_hearings)
+        end
+      end
+
+      context 'when should_save_court_hearings_in_nomis param is not present' do
```
Comment (willfish) [inline spec/requests/api/v1/court_hearings_controller_create_spec.rb:87]:
```suggestion
      context 'when should_save_in_nomis param is not present' do
```

**Pull request merged**